spec/support/rspec_cleanup.rb

```diff
@@ -16,6 +16,11 @@
     RequestStore.clear!
   end
 
+  config.append_after(:all) do
+    # Ensure models don't leak between test through RequestStore if it is used in after(:all)
+    RequestStore.clear!
```

Comment:
Now that you put this here: Doesn't line 16 already do exactly that?

Thinking about the `after(:all)`: Shouldn't that be `after(:each)`?

Reply:
Okay. Nevermind, maybe I understood now. Let me try to explain it and you tell me if I got it:

We already execute the clear after _each_ single example. However, since `after(:all)` callbacks run many times per test suite (at the end of each _group_), and since we have some `after(:all)` callbacks that might touch the `RequestStore`, these could then cause leaking.

About right?

Reply:
The problem is exactly in that `after(:each)` is not enough: the [test that brakes it](https://github.com/opf/openproject/blob/dev/modules/storages/spec/requests/api/v3/projects/projects_storages_filter_spec.rb) uses `after(:all)` from which `User.anonymous` is indirectly called leading it to be stored into `RequestStore` after it is cleaned by the `clear!` called from `after` on line 16